modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -18,56 +18,77 @@ import {useSelectModel} from './useSelectModel';
  * `useSelectInput` extends {@link useComboboxInput useComboboxInput}  and {@link useComboboxKeyboardTypeAhead useComboboxKeyboardTypeAhead} and adds type ahead functionality and Select-specific [keyboard support](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).
  */
 export const useSelectInput = composeHooks(
-  createElemPropsHook(useSelectModel)((model, ref, elemProps: {keySofar?: string} = {}) => {
-    const {localRef, elementRef} = useLocalRef<HTMLInputElement>(
-      useForkRef(ref as React.Ref<HTMLElement>, ref)
-    );
+  createElemPropsHook(useSelectModel)(
+    (model, ref, elemProps: {keySofar?: string; placeholder?: string} = {}) => {
+      const {localRef, elementRef} = useLocalRef<HTMLInputElement>(
+        useForkRef(ref as React.Ref<HTMLElement>, ref)
+      );
 
-    useResizeObserver({
-      ref: localRef,
-      onResize: data => {
-        if (model.state.visibility === 'visible') {
-          // Width of the Input + 2px border + 8px padding
-          const calculatedWidth = data.width ? data.width + 42 + 8 : 0;
-          model.events.setWidth(calculatedWidth);
-        }
-      },
-    });
-
-    return {
-      onKeyDown(event: React.KeyboardEvent) {
-        // Prevent the keys from being enter in the input
-        if (event.key !== 'Tab') {
-          event.preventDefault();
-        }
+      useResizeObserver({
+        ref: localRef,
+        onResize: data => {
+          if (model.state.visibility === 'visible') {
+            // Width of the Input + 2px border + 8px padding
+            const calculatedWidth = data.width ? data.width + 42 + 8 : 0;
+            model.events.setWidth(calculatedWidth);
+          }
+        },
+      });
 
-        // Select should open if Spacebar is typed and nothing has been typed AND the menu is hidden
+      React.useEffect(() => {
         if (
-          event.key === 'Spacebar' ||
-          (event.key === ' ' && model.state.visibility === 'hidden' && elemProps?.keySofar === '')
+          model.state.inputRef.current &&
+          model.state.items.length > 0 &&
+          model.state.selectedIds[0]
         ) {
-          model.events.show();
+          model.state.inputRef.current.value = model.navigation.getItem(
+            model.state.selectedIds[0],
+            model
+          ).textValue;
```

Comment:
Doing this skips the setter of the input which can cause problems for form libraries or anything using native `value` of the input. Both the Preview Select and the labs combobox use the native setter to change the value:

```ts
const input = model.state.inputRef.current;
const nativeInputValue = Object.getOwnPropertyDescriptor(Object.getPrototypeOf(input), 'value');
if (nativeInputValue && nativeInputValue.set) {
  nativeInputValue.set.call(input, model.navigation.getItem(
    model.state.selectedIds[0],
    model
  ).textValue)
}
```

This should trigger React Hook Form's internal updates so that if the user doesn't alter the value, the React Hook Form data representation is correct.